src/test/java/hw6/Test/JdiTest.java

```diff
@@ -0,0 +1,48 @@
+package hw6.Test;
+
+import com.epam.jdi.light.driver.WebDriverUtils;
+import com.epam.jdi.light.elements.init.PageFactory;
+import hw6.DataProviderJsonForMetalsColorPage;
+import hw6.JdiSite;
+import hw6.Json.JsonDataProvider;
+import hw6.complexelements.HeaderMenuItems;
+import org.testng.annotations.AfterSuite;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Test;
+
+@Test(dataProvider = "DataProviderJson", dataProviderClass = JsonDataProvider.class)
+public class JdiTest {
+
+    @BeforeSuite
+    public void beforeSuite() {
+        PageFactory.initSite(JdiSite.class);
+    }
+
+    @AfterSuite
+    public void afterSuite() {
+        WebDriverUtils.killAllSeleniumDrivers();
+    }
+
+    @Test
```

Comment:
Where is DataProvider ?

Reply:
C:\src\test\java\hw6\Json\JsonDataProvider -data provider here

I need to work with the names of methods and classes, because in my case they do not always correctly reflect the content

Reply:
If you mean the provider itself is described, then I rendered it in front of the class. This is the line 11